pages/Location/[id].js

```diff
@@ -0,0 +1,23 @@
+import React, { useState, useEffect } from "react";
+import { useRouter } from "next/router";
+import LocationCard from "../../components/LocationCard";
+import useLocalStorageState from "use-local-storage-state";
+
+export default function LocationDetail() {
+  const router = useRouter();
+  const { id } = router.query;
+
+  const [locations] = useLocalStorageState("favoriteLocations", {
+    defaultValue: [],
+  });
+  const [location, setLocation] = useState(null);
+
+  useEffect(() => {
+    if (id) {
+      const foundLocation = locations.find((loc) => loc.id === id);
+      setLocation(foundLocation);
+    }
+  }, [id, locations]);
```

Comment:
This works, but it's a misuse of the `useEffect` hook. You don't need it. You can leave it for now, but read up on how to avoid using `useEffect` in situations like yours [here](https://react.dev/learn/you-might-not-need-an-effect).

You can implement a fix later on when there's still time.

Reply:
Thanks for the link! I will fix that later then!